### DISCUSSION TITLE
Setting a disk cache size limit

### DISCUSSION THREAD

**Discussion by francis-io** [music-assistant/support#3253] [Feature requests and ideas]
I want to run Music Assistant on a device with limited local storage (Raspberry Pi CM4) with a max local storage size of 32GB. It would be great to be able to set a max local cache size (for example, 20 GB's), deleting the oldest accessed media first.

Other options to solve this problem:
* A cleanup script to run on a cron to delete the oldest files in the cache if the total size is over the set max limit (would MA gracefully redownload cached data if it suddenly goes missing?)
* Limiting the cache directory to only a certain size using Linux disk quotas
* Having a dedicated caching server, like Redis (but I expect this would be a big redesign of the app and logic)

Many thanks

**Comment by OzGav** [comment]
Thumbnail cache size limit is available in the 2.9.0 beta and later in MA SETTINGS >> SYSTEM >> METADATA

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
